cmake/Googletest.cmake

```diff
@@ -8,8 +8,6 @@ macro(fetch_googletest)
         if (NOT GTEST_ROOT)
             set(GTEST_ROOT "${GOOGLETEST_BUILD_ROOT}/googletest-install")
         endif()
-        find_package(GTest QUIET)
-        # At this point GTEST_FOUND is set to True in Release but False in Debug.
```

Comment:
The reason I moved it was because on the first time we build Gtest library, none of following variables are populated which is clearly a bug
```
GTEST_LIBRARY 
GTEST_LIBRARIES 
GTEST_MAIN_LIBRARY 
GTEST_MAIN_LIBRARIES 
```
If you look at line 51 we set GTEST_ROOT but after that none of Gtest variables get a chance to be filled out.
`set(GTEST_ROOT ${GOOGLETEST_BUILD_ROOT}/googletest-install CACHE path "GoogleTest installation root")`

I believe we need the call to `find_package(GTest QUIET)` both in line 11 and line 55 after GTEST_ROOT is set. 

Thanks for catching that.

Reply:
@seando-adsk  I want to revisit some of the logics we have here. Let's chat offline sometimes today.